features/admin.users.v1/components/wizard/steps/add-user-basic/add-user-basic.tsx

```diff
@@ -454,32 +449,202 @@ export const AddUserBasic: React.FunctionComponent<AddUserBasicProps> = ({
+                                tempProfileInfo.set(schema.name, customSchema);
+                            }
+                        } else {
+                            const subValue: SubValueInterface = userInfo[schemaName] &&
+                                Array.isArray(userInfo[schemaName]) &&
+                                userInfo[schemaName]
+                                    .find((subAttribute: MultiValueAttributeInterface) =>
+                                        subAttribute.type === schemaSecondaryProperty);
+
+                            if (schemaName.includes("addresses")) {
+                                // Ex: addresses#home.streetAddress
+                                const addressSubSchema: string = schema?.name?.split(".")[1];
+                                const addressSchemaArray: string[] = schemaName?.split("#");
+
+                                if (addressSchemaArray.length > 1) {
+                                    // Ex: addresses#home
+                                    const addressSchema: string = addressSchemaArray[0];
+                                    const addressType: string = addressSchemaArray[1];
+
+                                    const subValue: SubValueInterface = userInfo[addressSchema] &&
+                                        Array.isArray(userInfo[addressSchema]) &&
+                                        userInfo[addressSchema]
+                                            .find((subAttribute: MultiValueAttributeInterface) =>
+                                                subAttribute.type === addressType);
+
+                                    tempProfileInfo.set(
+                                        schema.name,
+                                        (subValue && subValue[addressSubSchema]) ? subValue[addressSubSchema] : ""
+                                    );
+                                } else {
+                                    tempProfileInfo.set(
+                                        schema.name,
+                                        subValue ? subValue.formatted : ""
+                                    );
+                                }
+                            } else {
+                                tempProfileInfo.set(
+                                    schema.name,
+                                    subValue ? subValue.value : ""
+                                );
+                            }
+                        }
+                    }
+                }
+            });
+
+            return tempProfileInfo;
+        }
+    };
+
     /*
     * This map the user data to schema using initial values of the wizard.
     * This is used to persist wizard data when the user clicks on the back button.
     */
-    useEffect(() => {
-        if (!isAttributeProfileForUserCreationEnabled) {
-            return;
+    const profileInfo: Map<string, string> = useMemo(() => {
```

Comment:
Update the `useEffect` to a `useMemo` to properly initialize the initial values at the initial rendering.